### DISCUSSION TITLE
No DbContext was found in assembly

### DISCUSSION THREAD

**Discussion by franciscoguerra** [balta-io/forum#82] [Tira Dúvidas] [answered]
Olá,
estou com dificuldade para resolver esse erro, quando vou usar o comando de migração do bando aparece essa mensagem -> No DbContext was found in assembly 'Todo2'. Ensure that you're using the correct assembly and that the type is neither abstract nor generic.

ja iniciei o projeto algumas vezes do inicio porém não foi solucionado. 
![erro-migrations database](https://user-images.githubusercontent.com/12381369/200315023-798d1464-41e9-4035-9273-00c3a83477a5.png)

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @franciscoguerra isto provavelmente ocorre por que seu projeto do EF não é um executável e sim uma classlib.
Você pode especificar o assembly na configuração da sua aplicação web..

```csharp
services.AddDbContext<ApplicationDbContext>(
    options =>
        options.UseSqlServer(
            Configuration.GetConnectionString("DefaultConnection"),
            x => x.MigrationsAssembly("WebApplication1.Migrations")));
```

Mais informações aqui: https://learn.microsoft.com/en-us/ef/core/managing-schemas/migrations/projects?tabs=dotnet-core-cli

**Reply by franciscoguerra** [reply]
Consegui criar as migrações e o banco, porém quando executei com o controller deu esse erro.

![erro2](https://user-images.githubusercontent.com/12381369/200720622-a41aad83-e4e2-4bd0-89e1-7c770f9f9d5c.png)

**Reply by andrebaltieri** [reply]
@franciscoguerra no caso, você precisa aceitar as opções no construtor do seu DbContext... pelo erro, está dizendo que não há um construtor que aceite o `DbContextOptions`.

```csharp
public MeuContexto(DbContextOptions<MeuContexto> options) : base(options)...
```
